rclrs/src/node/mod.rs

```diff
@@ -312,3 +305,57 @@ impl Node {
         domain_id
     }
 }
+
+impl NodeBuilder {
+    /// Creates new builder for Node.
+    pub fn new(name: &str, context: &Context) -> NodeBuilder {
+        NodeBuilder {
+            context: context.handle.clone(),
+            name: name.to_string(),
+            namespace: "/".to_string(),
+        }
+    }
+
+    /// Sets node namespace.
```

Comment:
I suggest to move the `Naming` documentation section from the `Node::new_with_namespace` function to the `Node` struct, to delete the `Node::new_with_namespace` function, and to add the following documentation here:

```suggestion
    /// Sets the node namespace.
    ///
    /// The node namespace needs to fulfill the criteria of
    /// [`rmw_validate_namespace()`][1], otherwise an error will be returned by the
    /// [`NodeBuilder::build`] function.
    ///
    /// A namespace without a leading forward slash is automatically changed to have a leading
    /// forward slash.
    ///
    /// See also the [section on node naming][2].
    ///
    /// # Example
    /// ```
    /// # use rclrs::{Context, NodeBuilder, RclReturnCode, NodeErrorCode};
    /// let context = Context::new([])?;
    /// let node_builder = NodeBuilder::new(&context, "my_node");
    /// // This is a valid namespace
    /// assert!(node_builder.namespace("/my/nested/namespace").build().is_ok());
    /// // This namespace contains invalid characters
    /// assert_eq!(
    ///     node_builder.namespace("/röböt").build(),
    ///     Err(RclReturnCode::NodeError(NodeErrorCode::NodeInvalidNamespace))
    /// );
    /// // This namespace starts with a number
    /// assert_eq!(
    ///     node_builder.namespace("/123/4").build(),
    ///     Err(RclReturnCode::NodeError(NodeErrorCode::NodeInvalidNamespace))
    /// );
    /// # Ok::<(), RclReturnCode>(())
    /// ```
    ///
    /// [1]: https://docs.ros2.org/latest/api/rmw/validate__namespace_8h.html
    /// [2]: crate::Node#naming
```

Reply:
The example won't compile yet – you probably need to at least derive `Clone` for `NodeBuilder` and add `.clone()` calls in there.